### PR TITLE
Set up initial file structure and public imports

### PR DESCRIPTION
**Pull request by jmh530**
My short-term plan is to put `dispersion`, `skewness`, `kurtosis`, and `quantile` in `./mir/descriptive.d` and `sweep`, `scale`, and `zScore` in `./mir/transform.d`. 

One thing I wasn't 100% on is should the folder paths be `./mir/math/package.d`, `./mir/math/descriptive.d`, `./mir/math/transform.d` to match the structure in `mir-algorithm`.

### CODE REVIEW

File: dub.sdl [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name "mir-stat"
+description "Dlang Statistical Package"
+
+authors "Ilya Yaroshenko" "John Michael Hall"
+copyright "Copyright © 2016 - 2019, Ilya Yaroshenko; see also information per file."
+license "BSL-1.0"
+
+dependency "mir-core" version=">=1.1.7"
```
Comment (9il) [inline dub.sdl]: don't need to define mir-core because mir-algorithm depends on it anyway.

File: dub.sdl [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+dependency "mir-algorithm" version=">=3.9.2"
+
+buildType "unittest" {
+	buildOptions "unittests" "debugMode" "debugInfo"
```
Comment (9il) [inline dub.sdl]: let use spaces everywhere

File: dub.selections.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+	"fileVersion": 1,
+	"versions": {
```
Comment (9il) [inline dub.selections.json]: this file should be removed

Comment (9il) [reply dub.selections.json]: also, needs meson.build, travis and circle ci, but they can be added in a separate mr

Comment (jmh530) [reply dub.selections.json]: I copied over the gitignore from mir-algorithm as well.

File: dub.sdl (hunk 2 above)
Comment (jmh530) [reply dub.sdl]: I had copied this from mir-algorithm so the tabs should be in there as well.

**Comment by 9il** [comment]
> One thing I wasn't 100% on is should the folder paths be `./mir/math/package.d`, `./mir/math/descriptive.d`, `./mir/math/transform.d` to match the structure in `mir-algorithm`.

`source/mir/stat/<filename>.d`

File: dub.sdl [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    dflags "-lowmem"
+}
+buildType "unittest-cov" {
+    buildOptions "unittests" "coverage" "debugMode" "debugInfo"
+    versions "mir_test"
+    dflags "-lowmem"
+}
+buildType "unittest-release" {
+    buildOptions "unittests" "releaseMode" "optimize" "inline" "noBoundsCheck"
+    versions "mir_test"
```
Comment (9il) [inline dub.sdl]: `mir_stat_test` everywhere

File: dub.sdl [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    dflags "-lowmem"
+}
+
+configuration "default" {
```
Comment (9il) [inline dub.sdl]: This package doesn't need configurations at all

File: dub.sdl [hunk 6]
```diff
@@ -0,0 +1,39 @@
+name "mir-stat"
+description "Dlang Statistical Package"
+
+authors "Ilya Yaroshenko" "John Michael Hall"
```
Comment (9il) [inline dub.sdl]: You can place you at the first here. It may be changed in the future depending on the author's contribution.

File: dub.sdl [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+copyright "Copyright © 2016 - 2019, Ilya Yaroshenko; see also information per file."
```
Comment (9il) [inline dub.sdl]: maybe `Copyright © 2020, Mir Stat Authors`

Comment (9il) [reply dub.sdl]: `Stat`!

**Pull request merged**